src/main/java/org/example/PerformWord.java

```diff
@@ -0,0 +1,44 @@
+package org.example;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.StringTokenizer;
+
+/*
+In this performWord it is used to search the given word present in file
+count the number of ccurance of word and pass the information to main thread
+by using implements Runnable and overriding the run()
+ */
+public class PerformWord implements Runnable {
+    private final String inputFilePath;
+    private final String searchWord;
+
+    public PerformWord(String inputFilePath, String searchWord) {
+        this.inputFilePath = inputFilePath;
+        this.searchWord = searchWord;
+    }
+
+    @Override
+    public void run() {
+        int matchingResultsCount = 0;
+        String inputDataFromFile;
+        try {
+            inputDataFromFile = new String(Files.readAllBytes(Paths.get(inputFilePath)));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
Here all the logic should be try catch block

Reply:
Updated has suggested.